sktree/stats/tests/test_forestht.py

```diff
@@ -0,0 +1,414 @@
+
+    # one must call `reset()` to make sure the test is run on a "new" feature set properly
+    with pytest.raises(RuntimeError, match="X must have 8 features"):
+        clf.statistic(iris_X, iris_y, metric="auc", max_fpr=limit)
+
+    clf.reset()
+    score = clf.statistic(iris_X, iris_y, metric="auc", max_fpr=limit)
+    assert score >= 0.8, "Failed with pAUC: {0} for max fpr: {1}".format(score, limit)
+
+    assert isinstance(clf.estimator_, HonestForestClassifier)
+
+
+@pytest.mark.parametrize(
+    "forest_hyppo",
+    [
+        FeatureImportanceForestRegressor(
+            estimator=RandomForestRegressor(
+                n_estimators=10,
+            ),
+            random_state=seed,
+        ),
+        FeatureImportanceForestClassifier(
+            estimator=RandomForestClassifier(
+                n_estimators=10,
+            ),
+            random_state=seed,
+            permute_per_tree=False,
+            sample_dataset_per_tree=False,
+        ),
+    ],
+)
+def test_forestht_check_inputs(forest_hyppo):
+    n_samples = 100
+    n_features = 5
+    X = rng.uniform(size=(n_samples, n_features))
+    y = rng.integers(0, 2, size=n_samples)  # Binary classification
+
+    # Test case 1: Valid input
+    forest_hyppo.statistic(X, y)
+
+    # Test case 2: Invalid input with different number of samples
+    X_invalid = np.random.rand(n_samples + 1, X.shape[1])
+    y_invalid = rng.integers(0, 2, size=n_samples + 1)
+    with pytest.raises(RuntimeError, match="X must have"):
+        forest_hyppo.statistic(X_invalid, y_invalid)
+
+    # Test case 3: Invalid input with different number of features
+    X_invalid = np.random.rand(X.shape[0], n_features + 1)
+    with pytest.raises(RuntimeError, match="X must have"):
+        forest_hyppo.statistic(X_invalid, y)
+
+    # Test case 4: Invalid input with incorrect y type target
+    y_invalid = np.random.rand(X.shape[0])
+    with pytest.raises(RuntimeError, match="y must have type"):
+        forest_hyppo.statistic(X, y_invalid)
+
+
+@pytest.mark.parametrize("backend", ["loky", "threading"])
+@pytest.mark.parametrize("n_jobs", [1, -1])
+def test_parallelization(backend, n_jobs):
+    """Test parallelization of training forests."""
+    n_samples = 100
+    n_features = 5
+    X = rng.uniform(size=(n_samples, n_features))
+    y = rng.integers(0, 2, size=n_samples)  # Binary classification
+
+    def run_forest(covariate_index=None):
+        clf = FeatureImportanceForestClassifier(
+            estimator=HonestForestClassifier(
+                n_estimators=10, random_state=seed, n_jobs=n_jobs, honest_fraction=0.2
+            ),
+            test_size=0.5,
+        )
+        pvalue = clf.test(X, y, covariate_index=[covariate_index], metric="mi")
+        return pvalue
+
+    out = Parallel(n_jobs=1, backend=backend)(
+        delayed(run_forest)(covariate_index) for covariate_index in range(n_features)
+    )
+    assert len(out) == n_features
```

Comment:
@sampan501 to my knowledge, any issues w/ joblib should be fixed or are the result of some other issues perhaps? 

Lmk if this unit-test sufficiently captures what the usage looks like in the power simulations.

Reply:
`n_jobs = 1` should be `n_jobs = n_jobs` in line 411, but otherwise good